src/commands.cpp

```diff
@@ -1998,173 +1998,281 @@ void CommandUserRequest::procresult()
     client->app->invite_result(e);
 }
 
-CommandPutUA::CommandPutUA(MegaClient* client, const char *an, const byte* av, unsigned avl)
+CommandPutUA::CommandPutUA(MegaClient* client, const char *an, const byte* av, unsigned avl, int ctag)
 {
-    cmd("up");
+    this->an = an;
+    this->av.assign((const char*)av, avl);
 
-    // if removing avatar, do not Base64 encode the attribute value
-    if (!strcmp(an, "+a") && !strcmp((const char *)av, "none"))
-    {
-        arg(an,(const char *)av, avl);
-    }
-    else
+    cmd("upv");
+
+    beginarray(an);
     {
-        arg(an, av, avl);
+        // if removing avatar, do not Base64 encode the attribute value
+        if (!strcmp(an, "+a") && !strcmp((const char *)av, "none"))
+        {
+            element((const char*)av);
+        }
+        else
+        {
+            element(av, avl);
+        }
+
+        const string *attrv = client->ownuser()->getattrversion(an);
+        if (attrv)
+        {
+            element(attrv->c_str());
```

Comment:
This could confuse servers because the number of elements in the array could change.